src/components/ContactForm/ContactForm.tsx

```diff
@@ -1,30 +1,139 @@
 import * as React from "react";
-
+import Recaptcha from "react-google-invisible-recaptcha";
+import { Button } from "../Button";
 import * as styles from "./ContactForm.module.scss";
 
 interface Props {
     style?: React.CSSProperties;
     className?: string;
 }
 interface State {
-    myStateValue?: boolean;
+    name: string;
+    email: string;
+    phone: string;
+    message: string;
 }
-
+const RECAPTCHA_KEY = process.env.GATSBY_SITE_RECAPTCHA_KEY;
 const INITIAL_STATE: State = {
-    myStateValue: true
+    name: "",
+    email: "",
+    phone: "",
+    message: ""
+};
+
+const encode = (data: { [key: string]: string }) => {
+    return Object.keys(data)
+        .map(
+            key => encodeURIComponent(key) + "=" + encodeURIComponent(data[key])
+        )
+        .join("&");
 };
 
 export class ContactForm extends React.PureComponent<Props, State> {
     public state = INITIAL_STATE;
+    public recaptcha?: Recaptcha;
+    public handleChange = (
+        e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) =>
+        this.setState({ [e.target.name]: e.target.value } as Pick<
+            State,
+            keyof State
+        >);
+
+    public handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        if (!this.recaptcha) {
+            console.error("Missing recaptcha");
+            return;
+        }
+        this.recaptcha.execute();
+    };
+
+    public handleCaptcha = () => {
+        console.log("handle captcha");
+        if (!this.recaptcha) {
+            console.error("Missing recaptcha");
+            return;
+        }
+        const captcha = this.recaptcha.getResponse();
+        console.log("got response", captcha);
+        this.sendMessage(captcha);
+    };
+    public sendMessage = async (captcha: string) => {
+        try {
+            await fetch("/", {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/x-www-form-urlencoded"
+                },
+                body: encode({
+                    "form-name": "contact",
+                    "g-recaptcha-response": captcha,
+                    ...this.state
+                })
+            });
+            alert("Sent message");
+        } catch (e) {
+            console.log("Error");
+        }
+    };
 
     public render() {
+        const { name, email, phone, message } = this.state;
         return (
-            <div
+            <form
+                onSubmit={this.handleSubmit}
                 className={[styles.component, this.props.className].join(" ")}
                 style={this.props.style}
+                data-netlify={true}
+                data-netlify-recaptcha={true}
+                name="contact"
             >
-                <h2>ContactForm</h2>
-            </div>
+                <div>
+                    <label htmlFor="name">Name</label>
+                    <input
```

Comment:
I believe that you need the id attribute for the label[for=*] to make the connection. `name` !== `id` and I think it is best to prefer id unless you want a non-unique reference